codeguru_profiler_agent/agent_metadata/aws_ec2_instance.py

```diff
@@ -27,12 +38,29 @@ def get_fleet_instance_id(self):
 
     @classmethod
     def __look_up_host_name(cls):
-        # The id of the fleet element. Eg. host name in ec2.
-        return http_get(url=EC2_HOST_NAME_URI).read().decode()
+        """
+        The id of the fleet element. Eg. host name in ec2.
+        """
+        return cls.__look_up_with_IMDSv2(EC2_HOST_NAME_URI)
 
     @classmethod
     def __look_up_instance_type(cls):
-        return http_get(url=EC2_HOST_INSTANCE_TYPE_URI).read().decode()
+        """
+        The type of the instance. Eg. m5.2xlarge
+        """
+        return cls.__look_up_with_IMDSv2(EC2_HOST_INSTANCE_TYPE_URI)
+
+    @classmethod
+    def __look_up_with_IMDSv2(cls, url):
+        return http_get(url=url,
+                        headers={EC2_METADATA_TOKEN_HEADER_KEY: cls.__look_up_ec2_api_token()}) \
```

Comment:
if we call both `__look_up_instance_type` and `__look_up_host_name` successively, we are going to get 2 different tokens, can you maybe update the `look_up_metadata` function to get a token once and reuse it?

Reply:
Okay, updated; thanks!